### PR TITLE
[RFR] [Feature] SimpleList redirect prop

### LINKED ISSUE #926: Allow changing what action happens on click in <SimpleList>
The `SimpleList ` right now redirects to Edit on click.

It would be nice to be able change the action (or even disable clicking).

(ie. i wanna it go to Show not Edit)

edit: probably something like:

```
import React from 'react';
import PropTypes from 'prop-types';
import { List, ListItem } from 'material-ui/List';
import { Link } from 'react-router-dom';

const tertiaryStyle = { float: 'right', opacity: 0.541176 };

const SimpleList = ({
    ids,
    data,
    basePath,
    primaryText,
    secondaryText,
    secondaryTextLines,
    tertiaryText,
    leftAvatar,
    leftIcon,
    rightAvatar,
    rightIcon,
    action,
}) => (
    <List>
        {ids.map(id => (
            <ListItem
                key={id}
                primaryText={
                    <div>
                        {primaryText(data[id], id)}
                        {tertiaryText &&
                            <span style={tertiaryStyle}>
                                {tertiaryText(data[id], id)}
                            </span>
                        }
                    </div>
                }
                secondaryText={secondaryText && secondaryText(data[id], id)}
                secondaryTextLines={secondaryTextLines}
                leftAvatar={leftAvatar && leftAvatar(data[id], id)}
                leftIcon={leftIcon && leftIcon(data[id], id)}
                rightAvatar={rightAvatar && rightAvatar(data[id], id)}
                rightIcon={rightIcon && rightIcon(data[id], id)}
                containerElement={action === 'none' ? 'span' : <Link to={`${basePath}/${id}${action ? `/${action}` : ''}`} />}
            />
        ))}
    </List>
);

SimpleList.propTypes = {
    ids: PropTypes.array,
    data: PropTypes.object,
    basePath: PropTypes.string,
    primaryText: PropTypes.func,
    secondaryText: PropTypes.func,
    secondaryTextLines: PropTypes.number,
    tertiaryText: PropTypes.func,
    leftAvatar: PropTypes.func,
    leftIcon: PropTypes.func,
    rightAvatar: PropTypes.func,
    rightIcon: PropTypes.func,
    action: PropTypes.string,
};

SimpleList.defaultProps = {
    action: '', // or default to 'none'?
};

export default SimpleList;
```

### PR DESCRIPTION
**Pull request by natrim**
to choose where to redirect on click
resolves #926

### CODE REVIEW

File: docs/List.md [hunk 1]
```diff
@@ -430,6 +430,44 @@ export const PostList = (props) => (
 );
 ```
 
+**Tip**: You can set what route the click on list item redirects to by setting the `redirect` prop.
+
```
Comment (djhi) [inline docs/List.md]: Please specify all the possible values and their effects

Comment (natrim) [reply docs/List.md]: it accept's the same as redirect props on edit and create

added the enums to doc

**Comment by fzaninotto** [comment]
The `redirect` name is incorrect, there is no redirection in that case. 

Besides, I'd prefer if you let the user override the `containerElement` altogether. So I suggest introducing an `itemContainerElement` prop instead of `redirect`.

**Comment by natrim** [comment]
@fzaninotto well i wanted to make it simple and similar to other components

https://github.com/marmelab/admin-on-rest/blob/master/src/mui/field/ReferenceField.js
is even simpler, with limited `linkType `

should i remake this pull with the `itemContainerElement ` ?

**Comment by natrim** [comment]
hm, i could probably just rename it to some other name and allow Element type

so the user could pass string for link
false for disabling
and element for replacing the container

**Comment by fzaninotto** [comment]
You're right, `linkType` already exists and it makes sense to reuse the same convention.

**Comment by natrim** [comment]
@fzaninotto should i the rename the `redirect` to `linkType` then ?
or some other change?

**Comment by fzaninotto** [comment]
renaming is fine. Just check that the semantics are the same as with the `ReferenceField` component.

**Comment by natrim** [comment]
what do you mean with `semantics`?

`ReferenceField ` accepts only: show, edit, true, false
    and defaults to edit

`SimpleList ` defaults to show
   - with the pull it accepts all redirectsTo supports
     - not 'true' though

**Comment by fzaninotto** [comment]
I mean that `SimpleList` should also accept `show`, `edit`, `true`, and `false`. It can default to `show`, though.

**Comment by natrim** [comment]
@fzaninotto renamed and changed to work same as in `ReferenceField `

**Comment by fzaninotto** [comment]
Thanks!

**Comment by natrim** [comment]
edition page?
i think it would be better to say: editing page

**Pull request merged**